src/kit/DataGrid/custom-renderers/utils.ts

```diff
@@ -105,7 +109,35 @@ export function drawTextWithEllipsis(
   x: number,
   y: number,
   maxWidth: number,
-): void {
+): TextMetrics {
   const ellipsisText = truncate(ctx, text, x, maxWidth);
   ctx.fillText(ellipsisText, x, y);
+  return ctx.measureText(ellipsisText);
+}
+
+export function drawTypeBadge(
+  ctx: CanvasRenderingContext2D,
+  theme: Theme,
+  typeName: 'number' | 'text' | 'date' | 'array' | 'unspecified',
+  x: number,
+  y: number,
+): void {
+  const tagFont = `bold 10px ${theme.fontFamily}`;
+  const bgColor = str2hsl(getComputedStyle(ctx.canvas).getPropertyValue('--theme-surface'));
+  const backgroundColor = hsl2str({
+    ...bgColor,
+    s: bgColor.s > 0 ? 70 : 0,
+  });
+  const textColor = getComputedStyle(ctx.canvas).getPropertyValue('--gdg-text-bubble');
```

Comment:
i think for the text color we might need to copy the text color logic from the badge as well. as mentioned before, we're using that color variable for the tags text border [source](https://github.com/determined-ai/determined-ui/blob/377bbbcd8254f0ae35b488cccf9d591ef14022ef/src/kit/DataGrid/custom-renderers/cells/tagsCell.tsx#L62-L65)

Reply:
`getComputedStyle(ctx.canvas).getPropertyValue('--gdg-text-bubble');` and `theme.textBubble` are equivalent. 
Did you mean `textDark` [code](https://github.com/determined-ai/hew/blob/377bbbcd8254f0ae35b488cccf9d591ef14022ef/src/kit/DataGrid/custom-renderers/cells/tagsCell.tsx#L68)?

with `textDark`

<img width="664" alt="image" src="https://github.com/user-attachments/assets/0523ac7b-048d-402c-9c5d-84327e54ee21">

<img width="689" alt="image" src="https://github.com/user-attachments/assets/f39a8100-83fc-4850-80b9-b140702a08f2">